### PR TITLE
Allow 1.2.3.dirty as non-orderable version

### PR DESCRIPTION
**Pull request by uschi2000**
This change is [Reviewable](https://reviewable.io/reviews/palantir/sls-packaging/202)

### CODE REVIEW

**Comment by uschi2000** [comment]
fyi @derenrich

File: sls-versions/src/test/java/com/palantir/slspackaging/versions/SlsProductVersionsTest.java [hunk 1]
```diff
@@ -23,11 +23,15 @@ public void testOrderableVersionDetection() {
...
         assertThat(SlsProductVersions.isNonOrderableVersion("2.0.0-beta1")).isTrue();
+        assertThat(SlsProductVersions.isNonOrderableVersion("2.0.0-beta1.dirty")).isTrue();
         assertThat(SlsProductVersions.isNonOrderableVersion("2.0.0-rc1")).isTrue();
+        assertThat(SlsProductVersions.isNonOrderableVersion("2.0.0-rc1.dirty")).isTrue();
         assertThat(SlsProductVersions.isNonOrderableVersion("2.0.0-beta1")).isTrue();
+        assertThat(SlsProductVersions.isNonOrderableVersion("2.0.0-beta1.dirty")).isTrue();
         assertThat(SlsProductVersions.isNonOrderableVersion("2.0.0-foo")).isTrue();
+        assertThat(SlsProductVersions.isNonOrderableVersion("2.0.0-foo.dirty")).isTrue();
         assertThat(SlsProductVersions.isNonOrderableVersion("2.0.0-foo-g20-gaaaaaa")).isTrue();
         assertThat(SlsProductVersions.isNonOrderableVersion("2.0.0-foo-g20-gaaaaaa.dirty")).isTrue();
```
Comment (markelliot) [inline sls-versions/src/test/java/com/palantir/slspackaging/versions/SlsProductVersionsTest.java]: How about a test that covers 1.0.0.dirty?

File: sls-versions/src/main/java/com/palantir/slspackaging/versions/SlsProductVersions.java [hunk 2]
```diff
@@ -6,7 +6,7 @@
  * Utility methods for checking whether version strings are valid SLS version strings.
  */
 public class SlsProductVersions {
-    private static final Pattern NON_ORDERABLE_VERSION = Pattern.compile("^[0-9]+\\.[0-9]+\\.[0-9]+-[a-z0-9-.]*$");
+    private static final Pattern NON_ORDERABLE_VERSION = Pattern.compile("^[0-9]+\\.[0-9]+\\.[0-9]+((-[a-z0-9-.]*)|(\\.dirty))$");
```
Comment (markelliot) [inline sls-versions/src/main/java/com/palantir/slspackaging/versions/SlsProductVersions.java]: Surprised this works -- would expect the or at the end to allow either but not both, and we explicitly want to allow both. Maybe clearer to replace "|(\\.dirty)" with "(\\.dirty)?" ?

Comment (uschi2000) [reply sls-versions/src/main/java/com/palantir/slspackaging/versions/SlsProductVersions.java]: note that `[a-z0-9-.]` accepts `.dirty` already

File: sls-versions/src/test/java/com/palantir/slspackaging/versions/SlsProductVersionsTest.java (hunk 1 above)
Comment (uschi2000) [reply sls-versions/src/test/java/com/palantir/slspackaging/versions/SlsProductVersionsTest.java]: added

File: sls-versions/src/main/java/com/palantir/slspackaging/versions/SlsProductVersions.java (hunk 2 above)
Comment (markelliot) [reply sls-versions/src/main/java/com/palantir/slspackaging/versions/SlsProductVersions.java]: ah, I missed that. I think I'd prefer to remove `.` support -- we added it just to support `.dirty` anyway.

Comment (markelliot) [reply sls-versions/src/main/java/com/palantir/slspackaging/versions/SlsProductVersions.java]: ie we want to allow `1.0.0.dirty` and `1.0.0-1-gabcdef.dirty` but did not really want to allow `1.0.0-rc.1`

**Pull request merged**